crates/miden-testing/tests/scripts/faucet.rs

```diff
@@ -138,9 +150,224 @@ fn faucet_contract_mint_fungible_asset_fails_exceeds_max_supply() -> anyhow::Res
+        decimals,
+        max_supply,
+        account_storage_mode,
+        auth_scheme,
+    )?;
+
+    // Verify the faucet account is properly created
+    assert_eq!(
+        faucet_account.account_type(),
+        miden_objects::account::AccountType::FungibleFaucet
+    );
+
+    // Verify the faucet can be extracted and has correct metadata
+    let faucet_component = BasicFungibleFaucet::try_from(faucet_account.clone())?;
+    assert_eq!(faucet_component.symbol(), token_symbol);
+    assert_eq!(faucet_component.decimals(), decimals);
+    assert_eq!(faucet_component.max_supply(), max_supply);
+
+    Ok(())
+}
+
+#[test]
+fn prove_new_faucet_execution_burn_fungible_asset_succeeds() -> anyhow::Result<()> {
+    // CONSTRUCT AND EXECUTE TX using new faucet that exercises the storage offset validation
+    // --------------------------------------------------------------------------------------------
+    let mut builder = MockChain::builder();
+    let faucet = builder.add_existing_faucet(Auth::BasicAuth, "TST", 200, Some(100))?;
+    let mut mock_chain = builder.build()?;
+
+    let fungible_asset = FungibleAsset::new(faucet.id(), 100).unwrap();
+
+    // Mint first to have some tokens to burn
+    let recipient = Word::from([0, 1, 2, 3u32]);
+    let tag = NoteTag::for_local_use_case(0, 0).unwrap();
+    let aux = Felt::new(27);
+    let note_execution_hint = NoteExecutionHint::on_block_slot(5, 6, 7);
+    let note_type = NoteType::Private;
+    let amount = Felt::new(100);
+
+    tag.validate(note_type).expect("note tag should support private notes");
+
+    let mint_tx_script_code = format!(
+        "
+            begin
+                # pad the stack before call
+                push.0.0.0 padw
+
+                push.{recipient}
+                push.{note_execution_hint}
+                push.{note_type}
+                push.{aux}
+                push.{tag}
+                push.{amount}
+                # => [amount, tag, aux, note_type, execution_hint, RECIPIENT, pad(7)]
+
+                call.::miden::contracts::faucets::basic_fungible::distribute
+                # => [note_idx, pad(15)]
+
+                # truncate the stack
+                dropw dropw dropw dropw
+            end
+            ",
+        note_type = note_type as u8,
+        recipient = recipient,
+        aux = aux,
+        tag = u32::from(tag),
+        note_execution_hint = Felt::from(note_execution_hint)
+    );
+
+    let mint_tx_script = ScriptBuilder::default().compile_tx_script(mint_tx_script_code)?;
+    let mint_tx_context = mock_chain
+        .build_tx_context(faucet.id(), &[], &[])?
+        .tx_script(mint_tx_script)
+        .build()?;
+    let mint_executed_transaction = mint_tx_context.execute_blocking()?;
+
+    prove_and_verify_transaction(mint_executed_transaction.clone())?;
+
+    let _minted_note = mint_executed_transaction.output_notes().get_note(0).clone();
+
+    // Now burn the minted asset
+    let note_script = "
```

Comment:
```suggestion
    let burn_note_script_code = "
```

Nit: Should we name this analogously to the mint script?